Initial address book bootstrap fails for ip_address_v4

### Description

Importer is failing with logs:
`Caused by: org.postgresql.util.PSQLException: ERROR: null value in column "ip_address_v4" of relation "address_book_service_endpoint" violates not-null constraint`

### Steps to reproduce

1. Check importer logs on integration.

### Additional context

_No response_

### Hedera network

other

### Version

v0.110.0-SNAPSHOT

### Operating system

None

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/addressbook/AddressBookServiceImpl.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/addressbook/AddressBookServiceImpl.java
@@ -19,6 +19,7 @@ package com.hedera.mirror.importer.addressbook;
 import static com.hedera.mirror.importer.config.CacheConfiguration.CACHE_ADDRESS_BOOK;
 import static com.hedera.mirror.importer.config.CacheConfiguration.CACHE_NAME;
 
+import com.google.protobuf.ByteString;
 import com.hedera.mirror.common.domain.addressbook.AddressBook;
 import com.hedera.mirror.common.domain.addressbook.AddressBookEntry;
 import com.hedera.mirror.common.domain.addressbook.AddressBookServiceEndpoint;
@@ -484,11 +485,13 @@ public class AddressBookServiceImpl implements AddressBookService {
 
     private AddressBookServiceEndpoint getAddressBookServiceEndpoint(
             ServiceEndpoint serviceEndpoint, long consensusTimestamp, long nodeId) throws UnknownHostException {
-        var ipAddressByteString = serviceEndpoint.getIpAddressV4();
-        String ip = StringUtils.EMPTY;
-        if (ipAddressByteString != null && ipAddressByteString.size() == 4) {
-            ip = InetAddress.getByAddress(ipAddressByteString.toByteArray()).getHostAddress();
+        var ipAddressByteString =
+                serviceEndpoint.getIpAddressV4() != null ? serviceEndpoint.getIpAddressV4() : ByteString.EMPTY;
+        if (ipAddressByteString.size() != 4) {
+            throw new IllegalStateException(String.format("Invalid IpAddressV4: %s", ipAddressByteString));
         }
+
+        var ip = InetAddress.getByAddress(ipAddressByteString.toByteArray()).getHostAddress();
         AddressBookServiceEndpoint addressBookServiceEndpoint = new AddressBookServiceEndpoint();
         addressBookServiceEndpoint.setConsensusTimestamp(consensusTimestamp);
         addressBookServiceEndpoint.setIpAddressV4(ip);

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/addressbook/AddressBookServiceImpl.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/addressbook/AddressBookServiceImpl.java
@@ -485,7 +485,7 @@ public class AddressBookServiceImpl implements AddressBookService {
     private AddressBookServiceEndpoint getAddressBookServiceEndpoint(
             ServiceEndpoint serviceEndpoint, long consensusTimestamp, long nodeId) throws UnknownHostException {
         var ipAddressByteString = serviceEndpoint.getIpAddressV4();
-        String ip = null;
+        String ip = StringUtils.EMPTY;
         if (ipAddressByteString != null && ipAddressByteString.size() == 4) {
             ip = InetAddress.getByAddress(ipAddressByteString.toByteArray()).getHostAddress();
         }
@@ -499,7 +499,7 @@ public class AddressBookServiceImpl implements AddressBookService {
             addressBookServiceEndpoint.setDomainName(serviceEndpoint.getDomainName());
         } else {
             // Setting domain_name to empty string here only until HIP 869 goes to mainnet
-            addressBookServiceEndpoint.setDomainName("");
+            addressBookServiceEndpoint.setDomainName(StringUtils.EMPTY);
         }
 
         return addressBookServiceEndpoint;

--- a/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/addressbook/AddressBookServiceImplTest.java
+++ b/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/addressbook/AddressBookServiceImplTest.java
@@ -65,6 +65,7 @@ import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.ValueSource;
 import org.junit.platform.commons.util.StringUtils;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.beans.factory.annotation.Value;
@@ -749,6 +750,38 @@ class AddressBookServiceImplTest extends ImporterIntegrationTest {
         assertEquals(2, addressBookRepository.count()); // bootstrap and new address book with service endpoints
         assertEquals(TEST_INITIAL_ADDRESS_BOOK_NODE_COUNT + addressBookEntries, addressBookEntryRepository.count());
         assertEquals(addressBookEntries * numEndpointsPerNode, addressBookServiceEndpointRepository.count());
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"1.0.0.0", ""})
+    void verifyAddressBookWithInvalidIp(String ip) {
+        entityProperties.getPersist().setNodes(true);
+
+        List<NodeAddress> nodeAddressList = new ArrayList<>();
+        int nodeAccountStart = 1;
+        NodeAddress.Builder nodeAddressBuilder = NodeAddress.newBuilder()
+                .setDescription("NodeAddressWithServiceEndpoint")
+                .setNodeAccountId(
+                        AccountID.newBuilder().setAccountNum(nodeAccountStart).build())
+                .setNodeCertHash(ByteString.copyFromUtf8(nodeAccountStart + "NodeCertHash"))
+                .setNodeId(nodeAccountStart - AddressBookServiceImpl.INITIAL_NODE_ID_ACCOUNT_ID_OFFSET)
+                .setRSAPubKey(nodeAccountStart + "RSAPubKey");
+
+        nodeAddressBuilder.addServiceEndpoint(ServiceEndpoint.newBuilder()
+                .setIpAddressV4(ByteString.copyFromUtf8(ip))
+                .setPort(52118)
+                .build());
+        nodeAddressList.add(nodeAddressBuilder.build());
+        NodeAddressBook.Builder nodeAddressBookBuilder =
+                NodeAddressBook.newBuilder().addAllNodeAddress(nodeAddressList);
+
+        byte[] addressBookBytes = nodeAddressBookBuilder.build().toByteArray();
+        update(addressBookBytes, 2L, false);
+
+        assertThat(addressBookRepository.count())
+                .isOne(); // only gets the bootstrap addressbook, cannot parse the one with service endpoints
+        assertThat(addressBookServiceEndpointRepository.count()).isZero();
+        entityProperties.getPersist().setNodes(false);
     }
 
     @Test

